include/aws/mqtt/client.h

```diff
@@ -307,6 +308,16 @@ void aws_mqtt_client_release(struct aws_mqtt_client *client);
 AWS_MQTT_API
 struct aws_mqtt_client_connection *aws_mqtt_client_connection_new(struct aws_mqtt_client *client);
 
+/**
+ * Creates a new MQTT311 connection object that uses an MQTT5 client under the hood
+ *
+ * \param[in] client    The mqtt5 client to create the connection from
+ *
+ * \returns a new mqtt (311) connection on success, NULL otherwise
```

Comment:
Nitpick: I might reword `returns a new mqtt (311) connection on success, NULL otherwise` to `returns a new MQTT311 connection object on success, NULL otherwise` for consistency with the first sentence.